### DISCUSSION TITLE
RLS preventing me from inserting into table

### DISCUSSION THREAD

**Discussion by Murkrage** [supabase/supabase#5542] [Questions]
Hi there,

I'm trying to insert into a table that's got a RLS policy setup for both `INSERT` and `SELECT`. When inserting it's telling me RLS is blocking me from doing so. I've tried inserting using `{returning: "minimal"}` but that didn't work either.

<img width="573" alt="Screenshot 2022-02-18 at 21 17 52" src="https://user-images.githubusercontent.com/6362631/154755987-d12ccdf3-06df-475a-a19e-95caa448727a.png">

The table has a column `user_id` which points to the users table in `auth`. The default value, upon creation, is `uid()`. I've also tried directly setting the `user_id` while inserting into the table, but that didn't do much unfortunately.

Temporarily removing the `SELECT` policy doesn't seem to do anything to help me insert, so it must have something to do with the first policy.

EDIT:
I forgot to add that I'm trying to do this server side using Next.js.

**Comment by blumenthaler** [comment]
@Murkrage did you ever solve for this? I am running into a similar issue where the `SELECT` policy blocks the `INSERT` action

**Reply by Murkrage** [reply]
Unfortunately not. It seems to me the authenticated check in the policy is broken, but I haven't verified this yet.

**Comment by ssteiger** [comment]
I'm having the same problem. Any ideas what could be done to fix this?
